docs/source/UIDocumentation.md

```diff
@@ -1,160 +1,238 @@
-Frontend Design 
-=======
-### Architecture
-```app.js``` hosts the main app and calls ```content.js```, which loads up the main app. This file contains a few functions:
-### ```content.js```
-(The top most bullet contains the name of the method and the inner bullets are params.)
-* ```openModal```
-* ```closeModal```
-* ```addNewLayer``` - invoked by ```handleClick``` and passed in a JS object with layer information. 
-  * layer
+Frontend Design
+========
+## Architecture
+(All javascript files described here can be found in the /ide/static/js/ directory of Fabrik)
+
+#### ```addCommentModal.js```
+(The bullets contain methods specific to the file along with their explanations.)
+
+```addCommentModal.js``` is responsible for adding comments into the interface. It has three methods:
+* ```handleClick``` - invoked on click on modal
+* ```addComment``` - adds comment modal to RTC
+* ```render``` - renders the comment modals
+
+***
+
+#### ```app.js```
+```app.js``` hosts the main app, but contains nothing else than a call for ```content.js```.
+***
+
+#### ```canvas.js```
+```content.js``` invokes an instance of ```canvas.js``` that contains the following methods:
+* ```allowDrop``` - allows element to be dropped down
+* ```clickLayerEvent``` - invoked on a click of a layer or on drag of that layer
+* ```hoverLayerEvent``` - invoked on a hover
+* ```scrollCanvas``` - scrolls canvas
+* ```clickCanvas``` - invoked on click
+* ```updateLayerPosition``` - changes layer positon based on event
+* ```mouseUpEvent``` - invoked when mouse key is up
+* ```connectionEvent``` - invoked on connection; checks for cycles and modifies layers
+* ```detachConnectionEvent``` - modifies layers
+* ```render``` - renders canvas
+* ```drop``` - invoked on layer drop onto canvas; checks for errors
+
+```canvas.js``` also contains the code that decides whether a node's line needs to be rerouted if it is cutting through another node.
+
+#### ```canvas.js```'s placement algorithm
```

Comment:
Can you change this to `visualisation algorithm` instead of `placement algorithm`